### DISCUSSION TITLE
Issue deploying win2019 adcs

### DISCUSSION THREAD

**Discussion by leJoPra** [Marshall-Hallenbeck/red_team_attack_lab#5] [Q&A] [answered]
Hello, 

I am having some trouble deploying the lab, especially the ADCS. It gets stuck at "configure AD CS certification authority via Install-ADCSCertificationAuthority". If i try to cancel, delete win2019-adcs, and restart vagrant up for the machine, it gets stuck usually at the last ansible task which is adding default certificate.
Any clue why this is happening?
Thanks

**Comment by Marshall-Hallenbeck** [comment, marked as answer]
@leJoPra So I had this same issue and I thought I fixed it. There's something weird happening during the ADCS deployment process where the Windows host isn't done setting up something related to the service.

Out of curiosity how many times have you tried to `vagrant up win2019-adcs`? I found that sometimes it works and sometimes it doesn't, so I had to `vagrant destroy win2019-adcs` and re-run it.

If you want to run the entire environment minus that host, just run a custom `vagrant up dc01 win2019-1 win10-1` etc.

I'll create an issue based on this and try to re-fix it (I thought I had it working).

**Reply by leJoPra** [reply]
Thanks for the answer.
In fact, I have tried many times to deploy it, sometimes it gets stuck later earlier in the web enrollment step and a few times it worked, but petitpotam attack (https://www.bussink.net/ad-cs-exploit-via-petitpotam-from-0-to-domain-domain/) never works (error generating certificate with ntlmrelayx). When I look in adcs manager, I get certificate failed. 
Can you try to do the same attack and tell me if it is related to my setup or the configuration of adcs itself ? 
Regards

**Reply by Marshall-Hallenbeck** [reply]
So I made sure that I could exploit it before, but it seems like I changed something and it messed it up.

Can you explain "when I look in ADCS manager I get certificate failed"? Can you post a screenshot of that?

Worst case right now, after running vagrant/ansible, even if it fails, go into the Server Manager via the UI and finish the ADCS installation manually (or make sure that everything is installed), then restart the computer. I know for sure this method works.

**Reply by leJoPra** [reply]
So here is what I get on ntlmrelayx :
![lab_ntlmrelayx](https://user-images.githubusercontent.com/57217348/135824775-b1238201-d402-4be8-bbb6-a76771b7a993.png)
And on the certification authority manager, in failed requests :
![lab_certif_enroll](https://user-images.githubusercontent.com/57217348/135825065-fb7885ee-b790-4860-8bd1-01e7f34a1484.png)
I have tried to add some more modules (Certificate Enrollment Policy Web Service) with the Server Manager UI but I get the same errors.

Error message on ADCS : 
`Active Directory Certificate Services denied request 3 because The requested certificate template is not supported by this CA. 0x80094800 (-2146875392 CERTSRV_E_UNSUPPORTED_CERT_TYPE).  The request was for CN=DC01$.  Additional information: Denied by Policy Module  0x80094800, The request was for a certificate template that is not supported by the Active Directory Certificate Services policy: Machine.`

**Reply by Marshall-Hallenbeck** [reply]
Sorry for the delay here, did you use https://github.com/ExAndroidDev/impacket.git & the `ntlmrelayx-adcs-attack` branch? 

Can you post the actual ntlmrelax command you used? From the looks of it, this is requesting a Machine ticket, but you need to request a Kerberos ticket like this:
`sudo python3 examples/ntlmrelayx.py -debug -smb2support --target http://10.0.1.14/certsrv/certfnsh.asp --adcs --template KerberosAuthentication`

**Reply by leJoPra** [reply]
Right! That was my mistake, forgot the `--template KerberosAuthentication` argument. Thanks for the troubleshooting and well done for the lab!

**Reply by Marshall-Hallenbeck** [reply]
You're welcome, I'm glad you're finding it helpful!

**Comment by avi8892** [comment]
@Marshall-Hallenbeck @leJoPra  It is impossible to install the ADCS server. it is always failed, event thought I have tried about 20 times.

**Reply by leJoPra** [reply]
Can you send the error message or the step where it fails ?
